src/commands/Info/rank.ts

```diff
@@ -40,12 +40,28 @@ const command: Command = {
 			currentXp = userDb.levels.levelXp,
 			messageCount = userDb.levels.messageCount
 
+		let currentXpFormatted: string,
+			xpNeededFormatted: string,
+			messagesFormatted: string
+
+		if (currentXp >= 1_000_000) currentXpFormatted = `${(currentXp / 1_000_000).toFixed(2)}${getString("million")}`
+		else if (currentXp >= 1000) currentXpFormatted = `${(currentXp / 1000).toFixed(2)}${getString("thousand")}`
+		else currentXpFormatted = `${currentXp}`
+
+		if (totalXp >= 1_000_000) xpNeededFormatted = `${(totalXp / 1_000_000).toFixed(2)}${getString("million")}`
+		else if (totalXp >= 1000) xpNeededFormatted = `${(totalXp / 1000).toFixed(2)}${getString("thousand")}`
+		else xpNeededFormatted = `${totalXp}`
+
+		if (messageCount >= 1_000_000) messagesFormatted = `${(messageCount / 1_000_000).toFixed(2)}${getString("million")}`
+		else if (messageCount >= 1000) messagesFormatted = `${(messageCount / 1000).toFixed(2)}${getString("thousand")}`
+		else messagesFormatted = `${messageCount}`
```

Comment:
Might as well just makes this into a function tbf

Reply:
Good idea, that way it can also be shared in levels.ts

Reply:
Only issue with it being in util is having to pass getString, so you can probably make the function in one of the files and export it to the other one

Reply:
Latest commit implements this functionality